File: Directory/38. Count and Say/Solution.cpp
/*
* @Author: yufeng
* @Date:   2018-10-25 18:11:42
* @Last Modified by:   yufeng
* @Last Modified time: 2018-10-25 18:54:32
* @Email: fzhiy270@sina.com
* @Blog: http://fzhiy.com/
*/

class Solution {
public:
    string countAndSay(int n) {
        string res = "1";
        for(int k=1; k<n; k++){
        	string t="";
        	for(int i=0, tmp; i<res.length(); i=tmp){
        		int cnt=1;
        		tmp=i;
        		while(tmp+1<res.length()&&res[tmp]==res[tmp+1]){
        			tmp++;
        			cnt++;
        		}
        		tmp++;
        		t+=to_string(cnt)+res[i];
        	}
        	res=t;
        }
        return res;
    }
};

File: Directory/38. Count and Say/Solution.py
# -*- coding: utf-8 -*-
# @Author: yufeng
# @Date:   2018-10-25 18:11:42
# @Last Modified by:   yufeng
# @Last Modified time: 2018-10-25 19:21:33
# @Email: fzhiy270@sina.com
# @Blog: http://fzhiy.com/

class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        res = "1"
        for k in range(1, n):
        	ch, t, cnt = res[0], "", 0
        	for ch2 in res:
        		if ch2 == ch:
        			cnt+=1
        		else :
        			t+=str(cnt)+ch
        			ch, cnt = ch2, 1
        	t+=str(cnt)+ch
        	res=t
        return res

n = 4
sol = Solution()
ans = sol.countAndSay(n)
print(ans)